lightning/src/routing/network_graph.rs

```diff
@@ -730,6 +740,11 @@ const MIN_SERIALIZATION_VERSION: u8 = 1;
 
 impl Writeable for NetworkGraph {
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		#[cfg(feature = "std")]
+		{
+			self.remove_stale_channels();
```

Comment:
This may be a bit of surprising side effect. Perhaps it should be documented on the struct?

Reply:
Yea, I waffled on it. I'm happy to remove it if you feel strongly, but it did seem nice to "do the obvious thing that users want" here. I suppose its awkward that there's no way to turn it off, but if users want a historical channel view they'll need to do their own thing anyway. In terms of things "just working", I think the only alternative would be to deref the `NetGraphMsgHandler` and call the timeout function regularly in the background-processor. For now I'll document it but I'm open to discussion either way.

Reply:
`NetGraphMsgHandler` (possibly renamed in the future :) seems like something that should be responsible for this. I'd even argue the method could be moved there as well. And then, as you said, called in the background processor.

Reply:
Wait, why would a message handler -> storage backend adapter be responsible for pruning? I'm fine with doing it in BP, but the message handler?

Reply:
Oh, I was just alluding to my opinion that message handling is simply one implementation detail (literally `impl RoutingMessageHandler`) of a broader gossip abstraction, of which the latter would also encompass pruning the network graph either by timestamp or monitoring the chain. Fine to keep this in `NetworkGraph` for now. Either way, it will be accessed through `NetGraphMsgHandler` in the background processor.

Reply:
> Oh, I was just alluding to my opinion that message handling is simply one implementation detail (literally impl RoutingMessageHandler) of a broader gossip abstraction, of which the latter would also encompass pruning the network graph either by timestamp or monitoring the chain. 

Huh, definitely feel like if we can keep these separate we should. But, sure, for now I'll move it to BP.